samples/matter/common/src/event_triggers/default_event_triggers.cpp

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <app/server/Server.h>
+#include <platform/nrfconnect/Reboot.h>
+
+#include "default_event_triggers.h"
+
+#include "event_triggers.h"
+#include <zephyr/logging/log.h>
+
+LOG_MODULE_DECLARE(app, CONFIG_CHIP_APP_LOG_LEVEL);
+
+using namespace Nrf::Matter;
+using namespace chip;
+
+namespace
+{
+k_timer sFactoryResetDelayTimer;
+k_timer sRebootDelayTimer;
+
+void FactoryResetDelayTimerTimeoutCallback(k_timer *timer)
+{
+	Server::GetInstance().ScheduleFactoryReset();
+}
+
+void RebootDelayTimerTimeoutCallback(k_timer *timer)
+{
+	DeviceLayer::SystemLayer().ScheduleLambda(
+		[] { DeviceLayer::Reboot(DeviceLayer::SoftwareRebootReason::kOther); });
```

Comment:
Should we emulate the ShutDown event is this case? Adding:
`PlatformMgr().HandleServerShuttingDown();
k_msleep(CHIP_DEVICE_CONFIG_SERVER_SHUTDOWN_ACTIONS_SLEEP_MS);`

should do the job.